site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,725 @@
+form of response message. You can also write your own response converter by implementing
+ResponseConverterFunction_ as follows. Also similar to RequestConverterFunction_, you can call
+``ResponseConverterFunction.fallthrough()`` when your response converter is not able to convert the result
+to an HttpResponse_.
+
+.. code-block:: java
+
+    public class MyResponseConverter implements ResponseConverterFunction {
+        @Override
+        public HttpResponse convertResponse(ServiceRequestContext ctx, Object result) {
+            if (result == MyObject.class) {
+                return HttpResponse.of(HttpStatus.OK,
+                                       MediaType.PLAIN_TEXT_UTF_8,
+                                       "Hello, %s!", result.processedName());
+            }
+
+            // To the next response converter.
+            return ResponseConverterFunction.fallthrough();
+        }
+    }
+
+You can annotate your service method and class as follows.
+
+.. code-block:: java
+
+    @ResponseConverter(MyResponseConverter.class)
+    public class MyAnnotatedServiceClass {
+
+        @Post("/hello")
+        public MyObject hello() {
+            // MyResponseConverter will be used to make a response.
+            // ...
+        }
+
+        @Post("/hola")
+        @ResponseConverter(MySpanishResponseConverter.class)
+        public MyObject hola() {
+            // MySpanishResponseConverter will be tried to convert MyObject to a response first.
+            // MyResponseConverter will be used if MySpanishResponseConverter fell through.
+            // ...
+        }
+    }
+
+Armeria supports :ref:`media_type_nego`. So you may want to get a negotiated media type in order to set
+a ``Content-Type`` header on your response. In this case, you can access it in your response converter
+as follows.
+
+.. code-block:: java
+
+    public class MyResponseConverter implements ResponseConverterFunction {
+        @Override
+        public HttpResponse convertResponse(ServiceRequestContext ctx, Object result)  {
+            MediaType mediaType = ctx.negotiatedProduceType();
+            if (mediaType != null) {
+                // Do something based on the media type.
+                // ...
+            }
+            // ...
+        }
+    }
+
+.. _configure-using-serverbuilder:
+
+Using ServerBuilder_ to configure converters and exception handlers
+-------------------------------------------------------------------
+
+You can specify converters and exception handlers using ServerBuilder_, without using the annotations
+explained in the previous sections::
+
+    sb.annotatedService(new MyAnnotatedServiceClass(),
+                        new MyExceptionHandler(), new MyRequestConverter(), new MyResponseConverter());
+
+When you specify exception handlers in a mixed manner like below, they will be evaluated in the following
+order commented. It is also the same as the evaluation order of the converters.
```

Comment:
It is also ... -> Please note that the same ordering rule applies for converters as well.